nifi-nar-bundles/nifi-hive-bundle/nifi-hive3-processors/src/main/java/org/apache/nifi/processors/hive/PutHive3Streaming.java

```diff
@@ -431,6 +431,10 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
                 if (rollbackOnFailure) {
                     throw new ProcessException(rrfe);
                 } else {
+                    log.error(
+                            "Failed to create {} for {}. rollbackOnFailure='{}' - routing to failure. Exception: {}",
```

Comment:
In this block `rollbackOnFailure` is always false. Also there are 4 `{}`params but only 3 arguments. I think you can get rid of the `rollbackOnFailure` one, but also instead of putting the Exception in the output string (which I think just calls `toString()` on it, you can pass the Exception `rrfe` as the third argument to `log.error()`.

Reply:
Applied all recommended changes.